### PR TITLE
Add nebulo.app for android

### PR DESCRIPTION
**Pull request by ignoramous**
[nebulo.app](https://nebulo.app/github) supports importing 1Hosts hostfile format.

### CODE REVIEW

**Comment by badmojr** [comment]
Hey,
I tested the app & wasn't impressed with how it handles domains with CNAMEs.
Though a domain is in a blocklist (e.g https://hosts.netlify.app/Pro/wildcards.txt), if it's CNAME resolves to a valid IP address, it automatically gets allowed.

**Comment by badmojr** [comment]
Also, since the app rejects duplicates by default; I recommend the wildcards variants of the lists as shown on the ss above.

**Review by badmojr — CHANGES_REQUESTED** [review]
Use the wildcards variants to reduce bandwidth use & address duplicates ommision.

**Comment by ignoramous** [comment]
Thanks a lot.

Pinged Daniel Wolf, creator of Nebulo... Expect them to reply anytime now.

**Comment by Ch4t4r** [comment]
Thanks @ignoramous for this PR! Nebulo currently uses a different format for wildcards, mainly because I couldn't find any source for how they are supposed to work a the time I added the feature. 
Speaking in terms of regexes Nebulo sees the wildcard `*.something.com` as regex `.*\.something\.com`, whereas you consider it to be `.*something\.com` (which in Nebulo equivalent would be `**something.com`).

Meaning if your wildcard rule would be `*tracking.india.miui.com` or `**india.miui.com` (or one of the other combinations..) it would work. As I am pretty sure the error is on my side though, I'll fix how Nebulo considers rules formatted that way.

**Comment by Ch4t4r** [comment]
Fixed for the versions in my Telegram channel & F-Droid repo. GPlay tomorrow.

**Comment by badmojr** [comment]
Hey @Ch4t4r .
It's blocking domains that are not on a given list; in my ss above https://hosts.netlify.app/Xtra/wildcards.txt

**Comment by Uranite** [comment]
> Hey @Ch4t4r .
> It's blocking domains that are not on a given list; in my ss above https://hosts.netlify.app/Xtra/wildcards.txt

same

**Comment by Ch4t4r** [comment]
This is because of the entry `*.b.com`. As I mentioned, Nebulo would previously match `something.b.com`, but not `b.com` with this rule (in Nebulo's own syntax this would be `**b.com`, where a single `*` means everything except a dot and `**` means everything, causing Nebulos rule set to be more flexible. In Nebulo you could do things like `ad*.adnetwork**` to match `ad1314.adnetwork.mycustomcname.com`).
What I did to facilitate this change is treating `*.something.com` as `**something.com` which I now see is wrong, as `**b.com` matches every domain which ends with `b.com`.

Long story short, nice catch, this could have gone ugly!

**Comment by badmojr** [comment]
That's quite a bug!
Glad that I've helped catch it.

**Comment by badmojr** [comment]
Hey @Ch4t4r .
Any update?

**Comment by badmojr** [comment]
Closed for luck of activity!
